static/src/js/components/AdminProjects/AdminProjectsList.tsx

```diff
@@ -0,0 +1,302 @@
+import React, {
+  useCallback,
+  useEffect,
+  useState
+} from 'react'
+import { useNavigate } from 'react-router-dom'
+import Pagination from 'rc-pagination'
+import localeInfo from 'rc-pagination/lib/locale/en_US'
+import Table from 'react-bootstrap/Table'
+import {
+  FaCaretUp,
+  FaCaretDown,
+  FaFilter
+} from 'react-icons/fa'
+import { gql, useQuery } from '@apollo/client'
+
+import Button from 'react-bootstrap/Button'
+import Col from 'react-bootstrap/Col'
+import Form from 'react-bootstrap/Form'
+import InputGroup from 'react-bootstrap/InputGroup'
+import Row from 'react-bootstrap/Row'
+
+import ADMIN_PROJECTS from '../../operations/queries/adminProjects'
+
+import EDSCIcon from '../EDSCIcon/EDSCIcon'
+import Spinner from '../Spinner/Spinner'
+
+import 'rc-pagination/assets/index.css'
+import './AdminProjectsList.scss'
+
+/**
+ * Interface defining the structure of a project user
+ */
+interface ProjectUser {
+  /** Unique identifier for the user */
+  id: string
+  /** URS username of the user */
+  ursId: string
+}
+
+/**
+ * Interface defining the structure of an admin project
+ */
+interface AdminProject {
+  /** Unique identifier for the project */
+  id: string
+  /** Human-readable project name */
+  name: string
+  /** Obfuscated unique identifier for the project */
+  obfuscatedId: string
+  /** Source path or query string for the project */
+  path: string
+  /** User who owns the project */
+  user: ProjectUser
+  /** ISO timestamp when the project was updated */
+  updatedAt: string
+  /** ISO timestamp when the project was created */
+  createdAt: string
+}
+
+/**
+ * Admin Projects GraphQL query result type
+ */
+interface AdminProjectsQueryData {
+  adminProjects: {
+    /** Array of admin projects returned from the API */
+    adminProjects: AdminProject[]
+    /** Total count of projects matching the query */
+    count: number
+  }
+}
+
+const AdminProjectsList = () => {
+  const navigate = useNavigate()
+
+  const [currentPage, setCurrentPage] = useState<number>(1)
+  const [searchType, setSearchType] = useState<string | undefined>('obfuscatedId')
+  const [searchValue, setSearchValue] = useState<string>('')
+  const [debouncedSearchValue, setDebouncedSearchValue] = useState<string>('')
+  const [sortKey, setSortKey] = useState<string | undefined>(undefined)
+
+  // Debounce the search value to avoid rapid requests
+  useEffect(() => {
+    const debounceHandler = setTimeout(() => {
+      setDebouncedSearchValue(searchValue)
+    }, 300)
```

Comment:
Do we have this same timeout for debounce in multiple places? If so, make it a constant.

Also, lodash has a [debounce](https://lodash.com/docs/4.17.15#debounce) function, would that be useful?